src/main.rs

```diff
@@ -200,7 +200,12 @@ fn main() {
     let result = match Command::from_args() {
         Command::Build { project_root } => command_build(project_root),
 
-        Command::Docs(Docs::Build { project_root, to }) => docs::command::build(project_root, to),
+        Command::Docs(Docs::Build { project_root, to }) => PathBuf::from(&project_root)
+            .canonicalize()
+            .map_err(|_| Error::UnableToFindProjectRoot {
```

Comment:
When can this error happen? And why does the path need to be canonicalized?

Reply:
No essential perhaps but we hit an error if provide a non-existent project directory as the first argument to `gleam docs build`:

```
$ gleam docs build non-existent-directory

error: Invalid project root

We were unable to find the project root:

  non-existent-directory
```
Without this we get an error when it tries to read `non-existent-directory/gleam.toml` which is relatively readable though not the clearest.

The motivation was this part:

```
            let relative_path = module_path
                .strip_prefix(&project_root)
```
Without canonicalising the `project_root`, that value is just `.` and so the `strip_prefix` doesn't work in the way that I hoped. That said, there might be a better approach to deriving a relative path than this.

The canonicalising does have side effects, eg. at the end of the `gleam docs build` output, it prints the absolute path to the docs folder instead of just `./gen/docs`.